surefire-booter/src/test/java/org/apache/maven/surefire/booter/ForkedBooterMockTest.java

```diff
@@ -362,6 +372,83 @@ public void run() throws Throwable
         }
     }
 
+    @Test
+    public void shouldAuthenticate() throws Exception
+    {
+        mockStatic( ForkedBooter.class );
+
+        doCallRealMethod()
+            .when( ForkedBooter.class, "lookupDecoderFactory", anyString() );
+
+        try ( final ServerSocketChannel server = ServerSocketChannel.open() )
+        {
+            if ( server.supportedOptions().contains( SO_REUSEADDR ) )
+            {
+                server.setOption( SO_REUSEADDR, true );
+            }
+
+            if ( server.supportedOptions().contains( TCP_NODELAY ) )
+            {
+                server.setOption( TCP_NODELAY, true );
+            }
+
+            if ( server.supportedOptions().contains( SO_KEEPALIVE ) )
```

Comment:
IMO you don't need these options in a test

Reply:
There are few of them.
@jon-bell 
@eolivelli 
Which options do we need, and which we need?

Reply:
Is is only a test case.
It is not real code that will be executed by users
You don't need any special option.

Reply:
@eolivelli 
@olamy 
Removed in test. Done!